app/page.tsx

```diff
@@ -1,15 +1,21 @@
 "use client";
-
-import SearchBar from "@/components/SearchBar";
 import React from "react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import SearchBar from "@/components/SearchBar";
+import SearchResult from "../components/SearchResult";
+
+const queryClient = new QueryClient();
 
 export default function Home() {
   const [searchValue, setSearchValue] = React.useState<string>("");
 
   return (
-    <main className="flex flex-col items-center justify-center ">
-      <SearchBar onChange={(value) => setSearchValue(value)} />
-      <h1 className="font-bold text-6xl mt-10">Home Page</h1>
-    </main>
+    <QueryClientProvider client={queryClient}>
```

Comment:
since you'll be using this context throughout, you'll want to move this higher up in the tree (your root `Layout.tsx`). Since context has to run in a client component, you want to structure it in a way to where your whole app doesn't become a client component.

[Here's an article](https://vercel.com/guides/react-context-state-management-nextjs) on how to create a `Providers.tsx` that allows you to add a global context without making everything a client component.

it's pretty confusing since it's context and next app directory stuff, so lmk if we need to pair on this